files/acl.py

```diff
@@ -21,6 +21,8 @@
 short_description: Sets and retrieves file ACL information.
 description:
      - Sets and retrieves file ACL information.
+notes:
+    - As of Ansible 2.0, this module only supports Linux distributions.
```

Comment:
Ah ah!

Not really, as everything this refactoring does is based off the `--test` option.

Once this PR makes it upstream, I will definitely work right away on a new PR to fix this. This implies a lot of code though, and this PR already does too much (add/fix capital-`X` support, other fixes, ...).
It would be very sad to see this PR delayed furthermore (initially opened more than a year ago). I promise I'll take care of BSD support right after that. I think I have shown good reactivity (if not always < 24h) every time I was asked, haven't I? :)